### ISSUE TITLE
Multi-GPU implementation

### ISSUE THREAD

**Issue by Lyken17** [bamos/densenet.pytorch#5]
Hi author

Thanks for sharing your code. I notice in README you said "Multi-gpu help wanted". If you are indicating data parallelism, then it can be implemented in several lines in pytorch using [nn.DataParallel](http://pytorch.org/docs/_modules/torch/nn/parallel/data_parallel.html#DataParallel). 

In your [train.py line 82](https://github.com/bamos/densenet.pytorch/blob/master/train.py#L83), simply modify code 

```python
    if args.cuda:
        net = net.cuda()
``` 
to
```python
    if args.cuda:
        net = net.cuda()
        net = nn.DataParallel(net, devices=[0,1,2,3])
```

can make whole model parallel.

**Comment by Lyken17** [comment]
I wonder, did you not implement it for some concerns? Like gradient correctness, numerical stability, convergence ... I just migrated from torch, and heard there are still some bugs in pytorch.

**Comment by varun-suresh** [comment]
I had the same question. @Lyken17 Did you train on multi-gpus?

**Comment by bamos** [comment]
Hi, I didn't try training on multiple GPUs. The issues @Lyken17 mentions can potentially happen, but I wouldn't expect hem to happen.

**Comment by Lyken17** [comment]
Hello @varun-suresh @bamos , though this reply is two months late, I want to tell you multi-gpus work as expect on pytorch. 

I use default setting in code. When using single gpu,  I get error rate of **5.01%**. When using 2 gpus, I get error rate of **4.67%.** Experiments on 3 and 4 gpus are on the way, I believe it will converge well. I will push a PR after experiments.

*  One GPU
![1 GPU](https://user-images.githubusercontent.com/7783214/28394573-c96ef62e-6ca2-11e7-9e05-3785a72f22ec.png)

*  Two GPUs
`python train.py --gpus 0,1  54374.10s user 3590.60s system 143% cpu 11:12:57.53 total`
![loss-error](https://user-images.githubusercontent.com/7783214/28400808-5bcdd298-6cca-11e7-8ef7-7c9e7a8a843c.png)

* Three GPUs
![loss-error](https://user-images.githubusercontent.com/7783214/28484708-cacf6580-6e28-11e7-864d-bea785bf25e0.png)

PS: I love 1080ti -- the most cost efficient card! I can buy more cards with the same cost.
